### PR TITLE
[HB-7211] Add public consent setter to non-TCF-compliant partner adapters

### PR DESCRIPTION
**Pull request by CB-RyanMcCormick**
Add function to adapter configuration class that allows publishers to set a boolean consent value to affect the partner SDK consent info.

### CODE REVIEW

**Review by chauduyphanvu — COMMENTED** [review]
logic lgtm but is this for 5.0 or now? either way we need the version bump and all that.

**Comment by CB-RyanMcCormick** [comment]
> logic lgtm but is this for 5.0 or now? either way we need the version bump and all that.

Yeah, the ticket is for mediation 5. Wanted to make sure this is what was intended before making additional changes.

Also, should we update the adapter versions to 5.x now?

**Comment by chauduyphanvu** [comment]
> > logic lgtm but is this for 5.0 or now? either way we need the version bump and all that.
> 
> Yeah, the ticket is for mediation 5. Wanted to make sure this is what was intended before making additional changes.
> 
> Also, should we update the adapter versions to 5.x now?

In that case we should PR against the 5.0.0 branch instead of against `main`. We can leave the version and stuff until closer to release date, I think.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 Note the first digit of every adapter version corresponds to the major version of the Chartboost Mediation SDK compatible with that adapter. 
 Adapters are compatible with any Chartboost Mediation SDK version within that major version.
 
+### 4.6.2.0.5
```
Comment (bichenwang) [inline CHANGELOG.md:6]: a bit disingenuous to put it on a version we won't release 🤔

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 2]
```diff
@@ -266,6 +266,28 @@ class HyprMXAdapter : PartnerAdapter {
         }
     }
 
+
+    /**
+     * Set HyprMX user's consent value using a boolean.
+     * This is for publishers to manually set the consent status.
+     * This uses CONSENT_GIVEN for true and CONSENT_DECLINED for false.
```
Comment (bichenwang) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:273]: GDPR_CONSENT_GRANTED/DENIED, no?

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 3]
```diff
@@ -266,6 +266,28 @@ class HyprMXAdapter : PartnerAdapter {
...
+     */
+    fun setGdpr(
+        context: Context,
+        applies: Boolean?,
+        consented: Boolean,
+    ) {
+        setGdpr(
+            context,
+            applies,
+            if (consented) GdprConsentStatus.GDPR_CONSENT_GRANTED else GdprConsentStatus.GDPR_CONSENT_DENIED,
```
Comment (bichenwang) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:287]: so technically applies=null = unknown, right? (though it likely doesn't matter; most of this is going away)

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt (hunk 2 above)
Comment (CB-RyanMcCormick) [reply HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:273]: Oops, yeah

**Review by bichenwang — APPROVED** [review]
oh oops confused applies with consented :P

**Pull request merged**